cmd/create.go

```diff
@@ -146,11 +152,42 @@ A full list of component types that can be deployed is available using: 'odo cat
 			os.Exit(1)
 		}
 
+		// If min-memory, max-memory and memory are passed, memory will be ignored as the other 2 have greater precedence.
+		// Emit a message indicating the same
+		if memoryMin != "" && memoryMax != "" && memory != "" {
+			fmt.Println("Flag `memory` will be ignored as `min-memory` and `max-memory` have been passed ")
+		}
+		if (memoryMin == "") != (memoryMax == "") && memory != "" {
+			fmt.Printf("Using memory %s for min and max limits.\n", memory)
+		}
+		if (memoryMin == "") != (memoryMax == "") && memory == "" {
```

Comment:
Is this a hard requirement? I mean it's definitely good practice to specify both, but is there a reason we have placed a hard requirement on having to specify both min and max fields or none at all?

Reply:
no min and max both or only memory(in which case internally we set min and max both to the same value)
only to avoid confusion
And setting both of them is mandated by openshift based on my experiments

Reply:
Understood :)